src/metabase/models/card.clj

```diff
@@ -175,6 +175,15 @@
       (log/info "Card references Fields in params:" field-ids)
       (field-values/update-field-values-for-on-demand-dbs! field-ids))))
 
+(defonce
+  ^{:doc "Atom containing a function used to check additional sandboxing constraints for Metabase Enterprise Edition. This is called as part of the `pre-update` method for a Card.
+
+  For the OSS edition, there is no implementation for this function -- it is a no-op. For Metabase Enterprise Edition,
+  the implementation of this function is
+  `metabase-enterprise.sandbox.models.group-table-access-policy/update-card-check-gtaps` and is installed by that
+  namespace."} pre-update-check-sandbox-constraints
+  (atom identity))
```

Comment:
Not sure if it would be cleaner or not, but I think we could also do this with multimethods: OSS gets `:default` and EE the implemented version.